### PR TITLE
Removes the horrible anim() proc and a bunch of spawns

### PR DESCRIPTION
**Pull request by ChangelingRain**
Also killed `/atom/movable/overlay` because oh WOW were all the uses a pile of crap
ninjacode get out

### CODE REVIEW

File: code/game/gamemodes/wizard/soulstone.dm [hunk 1]
```diff
@@ -220,12 +220,7 @@
 /obj/item/device/soulstone/proc/init_shade(mob/living/carbon/human/T, mob/U, vic = 0)
 	new /obj/effect/decal/remains/human(T.loc) //Spawns a skeleton
 	T.invisibility = INVISIBILITY_ABSTRACT
-	var/atom/movable/overlay/animation = new /atom/movable/overlay( T.loc )
-	animation.icon_state = "blank"
-	animation.icon = 'icons/mob/mob.dmi'
-	animation.master = T
-	flick("dust-h", animation)
```
Comment (ChangelingRain) [inline code/game/gamemodes/wizard/soulstone.dm]: flick() isn't fucking blocking except to other flicks and animate and such

File: code/__HELPERS/unsorted.dm [hunk 2]
```diff
@@ -496,25 +496,6 @@ Turf and target are seperate in case you want to teleport some distance from a t
 	var/y=arcsin(x/sqrt(1+x*x))
 	return y
 
-
-/proc/anim(turf/location,target as mob|obj,a_icon,a_icon_state as text,flick_anim as text,sleeptime = 0,direction as num)
```
Comment (ChangelingRain) [inline code/__HELPERS/unsorted.dm]: nearly every fucking use of this proc involved spawn(0)

**Comment by ShadowDeath6** [comment]
Your branch names are so long it breaks mobile browser :'^)

File: code/game/gamemodes/miniantags/abduction/machinery/pad.dm [hunk 3]
```diff
@@ -16,30 +16,27 @@
 	flick("alien-pad", src)
 	for(var/mob/living/target in loc)
 		target.forceMove(teleport_target)
-		spawn(0)
-			anim(target.loc,target,'icons/mob/mob.dmi',,"uncloak",,target.dir)
+		PoolOrNew(/obj/effect/overlay/temp/dir_setting/ninja, list(get_turf(target), target.dir))
```
Comment (Fox-McCloud) [inline code/game/gamemodes/miniantags/abduction/machinery/pad.dm]: I dunno why you guys still insist on using Pooling when it *still* has a net loss in performance, overall, even under 511.

**Pull request merged**